tests/test-db.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const test = require("tape-async");
+
+const DB = require("../db/DB");
+const getSha1 = require("../sha1-utils");
+
+test("getSubscriberByEmailAndToken accepts email and token and returns subscriber", async t => {
+  const testEmail = "unverifiedemail@test.com";
+  const testToken = "0e2cb147-2041-4e5b-8ca9-494e773b2cf0";
+  const subscriber = await DB.getSubscriberByEmailAndToken(testEmail, testToken);
+
+  t.ok(subscriber.email === testEmail);
+  t.ok(subscriber.verification_token === testToken);
+});
+
+test("getSubscribersByHashes accepts hashes and only returns verified subscribers", async t => {
+  const testHashes = [
+    "firefoxaccount@test.com",
+    "unverifiedemail@test.com",
+    "verifiedemail@test.com",
+  ].map(email => getSha1(email));
+  const subscribers = await DB.getSubscribersByHashes(testHashes);
+  for (const subscriber of subscribers) {
+    t.ok(subscriber.verified);
+  }
+});
+
+test("addSubscriberUnverifiedEmailHash accepts email and returns unverified subscriber with sha1 hash and verification token", async t => {
+  const testEmail = "test@test.com";
+  // https://stackoverflow.com/a/13653180
+  const uuidRE = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
```

Comment:
I think this is cool, although _mildly_ confused by that `-[1-5][0-9af]{3}-` bit in the middle there.

If I'm reading these links below correctly, it seems like that `[1-5]` may imply the version, and we could possibly just hardcode it as "4" since we're using UUIDv4, per https://github.com/mozilla/blurts-server/blob/069ea0bb3fa95347c46848d9be760716438fa82e/db/DB.js#L4

1. https://github.com/chriso/validator.js/blob/master/lib/isUUID.js#L16
1. https://gitlab.com/unwttng/uuid-v4-regex/blob/master/regex.js
1. https://github.com/mixer/uuid-validate/blob/master/index.js